sabnzbd/deobfuscate_filenames.py

```diff
@@ -217,64 +230,47 @@ def deobfuscate(nzo, filelist: List[str], usefulname: str):
-        if (
-            os.path.getsize(filename) > MIN_FILE_SIZE
-            and get_ext(filename) not in excluded_file_exts
-            and is_probably_obfuscated(filename)  # this as last test to avoid unnecessary analysis
-        ):
-            # Rename and make sure the new filename is unique
-            path, file = os.path.split(filename)
-            # construct new_name: <path><usefulname><extension>
-            new_name = get_unique_filename("%s%s" % (os.path.join(path, usefulname), get_ext(filename)))
-            logging.info("Deobfuscate renaming %s to %s", filename, new_name)
-            renamer(filename, new_name)
-            # find other files with the same basename in filelist, and rename them in the same way:
-            basedirfile, _ = os.path.splitext(filename)  # something like "/home/this/myiso"
-            for otherfile in filelist:
-                if otherfile.startswith(basedirfile + ".") and os.path.isfile(otherfile):
-                    # yes, same basedirfile, only different extension
-                    remainingextension = otherfile.replace(basedirfile, "")  # might be long ext, like ".dut.srt"
-                    new_name = get_unique_filename("%s%s" % (os.path.join(path, usefulname), remainingextension))
-                    logging.info("Deobfuscate renaming %s to %s", otherfile, new_name)
-                    # Rename and make sure the new filename is unique
-                    renamer(otherfile, new_name)
-                    nr_files_renamed += 1
-        else:
-            logging.debug("%s excluded from deobfuscation based on size, extension or non-obfuscation", filename)
+
+    # We pick the biggest file ... probably the most important file
+    filelist = sorted(filelist, key=os.path.getsize, reverse=True)
+    if filelist:
+        biggest_file = filelist[0]
+    else:
+        biggest_file = None
+
+    logging.debug("Deobfuscate inspecting biggest file%s", biggest_file)
+    if (
+        biggest_file
+        and big_small(filelist)
+        and get_ext(biggest_file) not in EXCLUDED_FILE_EXTS
+        and is_probably_obfuscated(biggest_file)
+        and os.path.isfile(biggest_file)
+    ):
+
+        # Rename and make sure the new filename is unique
+        path, file = os.path.split(biggest_file)
+        # construct new_name: <path><usefulname><extension>
+        new_name = get_unique_filename("%s%s" % (os.path.join(path, usefulname), get_ext(biggest_file)))
+        logging.info("Deobfuscate renaming %s to %s", biggest_file, new_name)
+        renamer(biggest_file, new_name)
+        nr_files_renamed += 1
+        # find other files with the same basename in filelist, and rename them in the same way:
+        basedirfile, _ = os.path.splitext(biggest_file)  # something like "/home/this/myiso"
+        for otherfile in filelist:
+            if otherfile.startswith(basedirfile + ".") and os.path.isfile(otherfile):
```

Comment:
Wouldn't it be better to not require the dot? That way it could also rename for instance xxx-sample.ext.

Reply:
Yes, great idea! I'll change it in the code, and I'll create a unit test for it.
Thanks.